javaparser-core/src/main/java/com/github/javaparser/printer/lexicalpreservation/LexicalDifferenceCalculator.java

```diff
@@ -146,6 +149,19 @@ private void calculatedSyntaxModelForNode(CsmElement csm, Node node, List<CsmEle
                 child = csmSingleReference.getProperty().getValueAsSingleReference(node);
             }
             if (child != null) {
+                // fix issue #2374
+                // Add node comment if needed (it's not very elegant but it works)
+                // We need to be sure that the node is an ExpressionStmt because we can meet
+                // this class definition
+                // a line comment <This is my class, with my comment> followed by
+                // class A {}
+                // In this case keyworld [class] is considered as a token and [A] is a child element
+                // So if we don't care that the node is an ExpressionStmt we could try to generate a wrong definition
+                // like this [class // This is my class, with my comment A {}]
+                if (node.getComment().isPresent() && node instanceof ExpressionStmt) {
```

Comment:
What is the defining characteristic of nodes like ExpressionStmt that make them need behaviour like this? Maybe we can make that a property of the node, so we can avoid node specific exceptional logic here.

Reply:
@jlerbsc ?

Reply:
I think it's more specific to lexical printer than expression statement.

Reply:
Okay, but do you think there are other nodes likes this? If not, it's okay.

Reply:
Sincerely, I don't know! I don't know enough JavaParser to answer this question. Maybe the author could answer it.

Reply:
Let's just merge it then :-)